components/brave_today/browser/publishers_controller.cc

```diff
@@ -172,6 +184,22 @@ void PublishersController::EnsurePublishersIsUpdating() {
         for (auto& observer : controller->observers_) {
           observer.OnPublishersUpdated(controller);
         }
+
+        if (base::FeatureList::IsEnabled(
+                brave_today::features::kBraveNewsV2Feature) &&
+            !missing_publishers_.empty()) {
+          controller->unsupported_publisher_migrator_->MigrateUnsupportedFeeds(
```

Comment:
I wonder if we should add a test for this, mainly to avoid breaking it in the future if we change pref formats.

Reply:
Yeah, so there's a test for the `unsupported_publisher migrator` but you're right, I don't think it will break if the format we store publishers in changes. I was having a bit of trouble setting up a unittest for `publishers_controller` because it depends on pretty much everything. 

I'll start having a dig into it, because we probably should have some test for it anyway :laugh

Reply:
Added the test, it was actually quite a bit easier than I thought!